web/src/Web.App/Services/ClaimsIdentifierService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Web.App.Domain;
+using Web.App.Identity.Models;
+using Web.App.Infrastructure.Apis;
+using Web.App.Infrastructure.Apis.Establishment;
+using Web.App.Infrastructure.Extensions;
+
+namespace Web.App.Services;
+
+public interface IClaimsIdentifierService
+{
+    Task<(string[] schools, string[] trusts)> IdentifyValidClaims(Organisation? organisation);
+}
+
+public class ClaimsIdentifierService(IEstablishmentApi api) : IClaimsIdentifierService
+{
+    public async Task<(string[] schools, string[] trusts)> IdentifyValidClaims(Organisation? organisation)
+    {
+        var schools = Array.Empty<string>();
+        var trusts = Array.Empty<string>();
+
+        if (organisation is { Category.Id: { } orgCategoryId })
+        {
+            switch (orgCategoryId)
+            {
+                case OrganisationCategories.SingleAcademyTrust:
+                case OrganisationCategories.MultiAcademyTrust:
+                    (schools, trusts) = await HandleTrustSchoolsAsync(organisation);
+                    break;
+                case OrganisationCategories.LocalAuthority:
+                    schools = await HandleLocalAuthoritySchoolsAsync(organisation);
+                    break;
+                default:
+                    var urn = organisation.UrnValue.ToString();
```

Comment:
nitpick: For consistency, could the `schools` and `trusts` Tuple be returned from `HandleAcademyTrustAsync()` as per the other branches in the `switch`? i.e. move lines 33-34 into `HandleAcademyTrustAsync()`

Reply:
yes that reads better, updated

Reply:
Based on that change though I'm not sure if that method name is suitable...